cmd/localCI/github.go

```diff
@@ -208,16 +209,23 @@ func (g *Github) downloadPullRequest(pr int, workingDirectory string) (string, e
 		return "", fmt.Errorf("failed to create project directory %s", err)
 	}
 
+	var stderr bytes.Buffer
+
 	cmd := exec.Command("git", "clone", g.url, ".")
 	cmd.Dir = projectDirectory
+	cmd.Stderr = &stderr
 	if err := cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to clone project %s", err)
+		ciLog.Errorf("%s", stderr.String())
+		return "", fmt.Errorf("failed to run git clone %s", err)
 	}
 
+	stderr.Reset()
 	cmd = exec.Command("git", "pull", "--no-edit", "origin", fmt.Sprintf("pull/%d/head", pr))
 	cmd.Dir = projectDirectory
+	cmd.Stderr = &stderr
 	if err := cmd.Run(); err != nil {
-		return "", fmt.Errorf("failed to pull request %s", err)
+		ciLog.Errorf("%s", stderr.String())
+		return "", fmt.Errorf("failed to run git pull %s", err)
```

Comment:
it was my first option, unfortunately sometimes the error message is big, I prefer to log it and return an error